### ISSUE TITLE
Stacking Popup of same type

### ISSUE THREAD

**Issue by dentvii** [Mijick/Popups#123]
#### Problem
Currently, it is not possible to stack popups of the same type. For instance, displaying multiple `TopNotification` popups simultaneously is not feasible because they all share the same identifier.

#### Code Example
```swift
public extension Popup {
    var id: String { .init(describing: Self.self) }
    var body: V { createContent() }

    func configurePopup(popup: Config) -> Config { popup }
}
```

#### Issue Details
Due to the same identifier (`TopPopup_Notification`), these messages can only replace each other rather than stack.

#### Proposed Solution
The `id` of the `Popup` should be a property of the caller, making it an optional description of self. This change would allow for multiple notifications of the same type to be displayed concurrently. 

#### Benefits of Custom Popup IDs
1. **Stacking Multiple Popups**: Enables multiple notifications of the same type to be displayed simultaneously.
2. **Targeted Removal**: Allows a specific popup to be removed from the stack.
3. **Existence Check**: Facilitates checking for the existence of a specific popup on the stack before taking action.
4. **Maintaining Current Behavior**: Keeps the current behavior intact while allowing for multiple instances of the same type.

#### Visual Example
<img width="683" alt="Popup Example" src="https://github.com/user-attachments/assets/d460c010-4641-449f-8202-a6f15657a5f8">
This has a while loop on this message on the demo, but only one message is shown. 

#### Expected Behavior
<img width="503" alt="image" src="https://github.com/user-attachments/assets/136820da-242c-4ccc-8ff1-f2a5fa24a585">
Messages of same Notification Type are reused and stacked. 

### Conclusion
Modifying the `id` property to be customizable will enhance the flexibility and functionality of the popup system, allowing for a more dynamic and responsive user experience.

**Comment by FulcrumOne** [comment]
Hey @dentvii,

Thanks for this post. It does indeed make sense. I'll try to squeeze this feature into the current sprint, but no promises, as I need to finally finish the new library I put on hold about two months ago (so in the worst case, this will be implemented at the end of August) 😅

Thanks for the ticket and have a nice day,
Tomasz

**Comment by jamesyorke** [comment]
Just adding to this as i'm also hitting this issue for a feature!

Making the `ID` property public/ with public initialisers should resolve for us?

**Comment by praveenperera** [comment]
I think i'm having the same issue, the loading popup here does not get replaced

```swift
    @MainActor
    private func startLoading() {
        MiddlePopup(state: .loading)
            .showAndStack()
    }

    @MainActor
    private func completeLoading(_ state: PopupState) {
        print("loading completed: \(state)")
        MiddlePopup(state: state)
            .showAndReplace()
    }
```

**Comment by FulcrumOne** [comment]
Hey, this feature is in our [backlog](https://mijick.notion.site/c95fe641ff684561a4523b9570113516?v=46960f9a652147d4a12fc9dd36cbd4fc) and will be implemented next month.

**Comment by praveenperera** [comment]
Cool thanks! And thanks for the great library, it’s much nicer to use than the pure state driven SwiftUI popup view libraries.

**Comment by FulcrumOne** [comment]
@jamesyorke, @dentvii, @praveenperera,

Just letting you know that the feature is already implemented in the branch **patch-3.0.0**. It can be used as follows:
![CleanShot 2024-10-04 at 15 23 12](https://github.com/user-attachments/assets/0030a8da-1b46-42e9-840b-fc7e3da54a7e)

I would definitely advise to not use the branch on prod, but you can test the changes if you have some time.
I plan to publish a pre-release version on **16 October** and launch it **7-10 days later**.

**Comment by FulcrumOne** [comment]
Hey @jamesyorke, @dentvii, @praveenperera

Just letting you know that I've finished the update. There are some changes to the API and the GitHub documentation (with migration guidelines) is not yet ready, but the library has extensive documentation with some code examples that I hope will help you understand the new features.
![CleanShot 2024-10-16 at 01 51 18](https://github.com/user-attachments/assets/745a9dc8-1c75-4fe4-b2b4-cb037c6d8c70)
![CleanShot 2024-10-16 at 01 52 14](https://github.com/user-attachments/assets/fd703b1c-cf4e-4cde-a664-5260ed0dd5e1)


Let me know if you need any help and have a nice day,
Tomasz
